lib/mapObjects/CRewardableObject.cpp

```diff
@@ -107,6 +108,10 @@ void CRewardableObject::onHeroVisit(const CGHeroInstance *h) const
 					case Rewardable::SELECT_RANDOM: // give random
 						grantRewardWithMessage(*RandomGeneratorUtil::nextItem(rewards, cb->gameState()->getRandomGenerator()), true);
 						break;
+					case Rewardable::SELECT_ALL: // give all rewards
+						for(auto i : rewards)
+							grantRewardWithMessage(i, i == rewards.size() - 1);
```

Comment:
Please check whether this mode works correctly if:
- multiple rewards levelup hero
- multiple rewards give creatures and hero has full army